src/config.nim

```diff
@@ -48,48 +48,62 @@ proc selfChalkSetSubKey*(key: string, subKey: string, subValue: Box) =
   value[subKey] = subValue
   selfChalkSetKey(key, pack(value))
 
-proc filterByTemplate*(dict: ChalkDict, p: MarkTemplate | ReportTemplate): ChalkDict =
+proc filterByTemplate*(dict: ChalkDict, p: AttrScope): ChalkDict =
   result = ChalkDict()
   for k, v in dict:
-    if k in p.keys and p.keys[k].use:
-      result[k] = v
+    let pKeysOpt = getObjectOpt(p, "key")
+    if pKeysOpt.isSome():
+      let pKeys = pKeysOpt.get()
+      if getObjectOpt(pKeys, k).isSome() and get[bool](pKeys, k & ".use"):
+        result[k] = v
 
-proc getOutputConfig*(): OutputConfig =
-  return chalkConfig.outputConfigs[getBaseCommandName()]
+proc getOutputConfig*(): AttrScope =
+  return getObject(getChalkScope(), "outconf." & getBaseCommandName())
 
-template getMarkTemplate*(): MarkTemplate =
+template getMarkTemplate*(): AttrScope =
   let
-    outconf  = chalkConfig.outputConfigs[getBaseCommandName()]
-    tmplName = outconf.mark_template
+    outconf  = getOutputConfig()
+    tmplName = get[string](outconf, "mark_template")
 
-  chalkConfig.markTemplates[tmplName]
+  getObject(getChalkScope(), "mark_template." & tmplName)
 
-template getReportTemplate*(): ReportTemplate =
+template getReportTemplate*(): AttrScope =
   let
-    outconf  = chalkConfig.outputConfigs[getBaseCommandName()]
-    tmplName = outconf.report_template
+    outconf  = getOutputConfig()
+    tmplName = get[string](outconf, "report_template")
 
-  chalkConfig.reportTemplates[tmplName]
+  getObject(getChalkScope(), "report_template." & tmplName)
 
 template forceReportKeys*(keynames: openarray[string]) =
   let templateRef = getReportTemplate()
 
+  # Create the "key" section if required.
+  if getObjectOpt(templateRef, "key").isNone() and keynames.len > 0:
+    discard attrLookup(templateRef, ["key"], 0, vlSecDef)
```

Comment:
can we use keyword args for `0` and `vlSecDef`? I have no idea what they mean in this context. same in other `attrLookup` usage which create a section